server/metadata/tenant.go

```diff
@@ -201,12 +201,28 @@ func (m *TenantManager) GetNamespace(ctx context.Context, namespaceName string,
 	m.RUnlock()
 
 	m.Lock()
-	defer m.Unlock()
 	if tenant, found := m.tenants[namespaceName]; found {
 		return tenant.namespace, nil
 	}
-	// this will never create new namespace
-	// this is for reading namespaces from storage into cache
+	m.Unlock()
```

Comment:
this won't work, you need to release the lock after populating the map.

Reply:
moved after it is done reading. Please review this one more time.